### ISSUE TITLE
Download Windows dependencies on the fly

### ISSUE THREAD

**Issue by PeterDaveHello** [leonsoft-kras/transmisson-remote-gui#1013]
The files under [`setup/win/openssl`](https://github.com/leonsoft-kras/transmisson-remote-gui/tree/master/setup/win/openssl) take about 5.6MB, and we currently package them in the Windows client installer, which is not efficient, the better method would be to download them on the fly, if and only if the users need it(lack of the dependency on the machine), so that we can save the size of installer, the bandwidth and time. Just like what we are doing with the flags images, no need to distribute them every single release.

**Comment by PeterDaveHello** [comment]
@leonsoft-kras not sure if you'd like to take a look?

**Comment by leonsoft-kras** [comment]
We need somehow to fix the installer.
In addition, if "dependencies" are not available, display a message with the address for downloading.

**Comment by PeterDaveHello** [comment]
Maybe also in the application (I mean the message, instead of crash).

**Comment by leonsoft-kras** [comment]
The setup.iss file must be fixed. For example, depending on the function VCRedistNeedsInstall do the loading from the Internet. If the Internet is not available to report the download address.
Inno5 should be studied.

**Comment by PeterDaveHello** [comment]
Landed on https://github.com/leonsoft-kras/transmisson-remote-gui/releases/tag/v5.11.0

**Comment by PeterDaveHello** [comment]
I think OpenSSL part should also be done.

**Comment by PeterDaveHello** [comment]
I don't want to pack OpenSSL in the installer.

**Comment by leonsoft-kras** [comment]
What do we get?
We will make the distribution kit 500 kb less?

**Comment by PeterDaveHello** [comment]
Yes, that's not necessary at all. Also, it's not a good idea to host those files in git repository.

**Comment by leonsoft-kras** [comment]
However, this ensures that anyone can download the project and create a fully working application. 
An external link can become void tomorrow.

Downloading vcredist was the right decision, but I do not see any reason to do the same with OpenSSL.

**Comment by PeterDaveHello** [comment]
There is no need to download it everytime, OpenSSL doesn't update so fast, and we didn't update the OpenSSL dll so fast, big part of the space and bandwidth is a wasting.

**Comment by PeterDaveHello** [comment]
Also, if the users already installed the OpenSSL library, there is no need to have it here.

**Comment by leonsoft-kras** [comment]
Why go far? Until recently, this project was on the website of Google, which was closed.
The library does not occupy much space. )))

Remember how many problems there were with Linux when the OpenSSL did not work. An instruction was written that needed to be installed.

Total, I think this is not a primary problem. I have many other tasks)))

**Comment by PeterDaveHello** [comment]
I don't mean that you need to do it, make it open and every one can do it. The Linux problem is not a real problem, every problem comes from a not good enough implementation, we can do it good in the correct way.
